File: src/com/leetcode/Easy/SearchInsertPosition.java
package com.leetcode.Easy;

// 35. Search Insert Position
public class SearchInsertPosition {

    public int searchInsert(int[] nums, int target) {
        int index = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == target) {
                return i;
            }

            if (nums[i] < target) {
                index = i + 1;
            }
        }
        return index;
    }
}

File: JavaScript/Easy/SearchInsertPosition.js
/**
 * 35. Search Insert Position
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
  if (nums.length == 0 || target < nums[0]) return 0;

  if (target > nums[nums.length - 1]) return nums.length;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == target) return i;
  }

  let index = 0;
  for (let i = 1; i < nums.length; i++) {
    if (target > nums[i - 1] && target < nums[i]) index = i;
  }
  return index;
};
